ConnectorAutocadCivil/ConnectorAutocadCivil/Utils.cs

```diff
@@ -254,95 +254,193 @@ public static bool Visible(this DBObject obj)
         return null;
     }
   }
 
-  public static void SetPropertySets(this Entity entity, Document doc, List<Dictionary<string, object>> propertySetDicts)
+  public static PropertySetDefinition CreatePropertySet(Dictionary<string,object> propertySetDict, Document doc)
   {
-    // create a dictionary for property sets for this object
-    var name = $"Speckle {entity.Handle} Property Set";
-    int count = 0;
-    foreach (var propertySetDict in propertySetDicts)
-    {
-      // create the property set definition for this set.
-      var propSetDef = new PropertySetDefinition();
-      propSetDef.SetToStandard(doc.Database);
-      propSetDef.SubSetDatabaseDefaults(doc.Database);
-      var propSetDefName = name += $" - {count}";
-      propSetDef.Description = "Property Set Definition added with Speckle";
-      propSetDef.AppliesToAll = true;
-
-      // Create the definition for each property
-      foreach (var entry in propertySetDict)
+    PropertySetDefinition propSetDef = new();
+    propSetDef.SetToStandard(doc.Database);
+    propSetDef.SubSetDatabaseDefaults(doc.Database);
+    propSetDef.Description = "Property Set Definition added with Speckle";
+    propSetDef.AppliesToAll = true;
+
+    // Create the definition for each property
+    foreach (var entry in propertySetDict)
+    {
+      var propDef = new PropertyDefinition();
+      propDef.SetToStandard(doc.Database);
+      propDef.SubSetDatabaseDefaults(doc.Database);
+      propDef.Name = entry.Key;
+      if (GetPropertySetType(entry.Value) is Autodesk.Aec.PropertyData.DataType dataType)
       {
-        var propDef = new PropertyDefinition();
-        propDef.SetToStandard(doc.Database);
-        propDef.SubSetDatabaseDefaults(doc.Database);
-        propDef.Name = entry.Key;
-        var dataType = GetPropertySetType(entry.Value);
-        if (dataType != null)
+        propDef.DataType = dataType;
+      }
+
+      propDef.DefaultData = entry.Value;
+      propSetDef.Definitions.Add(propDef);
+    }
+
+    return propSetDef;
+  }
+
+  /// <summary>
+  /// Finds a property set by its ObjectId on a given object.
+  /// </summary>
+  /// <param name="obj">The object to find the property set on.</param>
+  /// <param name="propertySetId">The property set ObjectId to find on the object.</param>
+  /// <returns> True if the property set with the given ObjectId was found, or false otherwise. </returns>
+  public static bool ObjectHasPropertySet(DBObject obj, ObjectId propertySetId)
+  {
+    ObjectId temporaryId = ObjectId.Null;
+
+    try
+    {
+      // attempt to retrieve the property set.
+      // this will throw if the property set does not exist on the object.
+      // afaik, trycatch is necessary because there is no way to preemptively check if the set already exists.
+      temporaryId = PropertyDataServices.GetPropertySet(obj, propertySetId);
+    } 
+    catch (Autodesk.AutoCAD.Runtime.Exception)
+    {
+      // More than likely eKeyNotFound exception.
```

Comment:
Do we not want to make sure this IS what we are seeing? What if it were not?

Reply:
No exceptions were documented on the `GetPropertySet` method, from previous testing the runtime exception had an `eKeyNotFound` message. Added !IsFatal() here